src/App.tsx

```diff
@@ -1,52 +1,84 @@
+import { NotFound } from './components/NotFound';
+import { SearchLink } from './components/SearchLink';
+import { tabs } from './constats';
 
-// const tabs = [
-//   { id: 'tab-1', title: 'Tab 1', content: 'Some text 1' },
-//   { id: 'tab-2', title: 'Tab 2', content: 'Some text 2' },
-//   { id: 'tab-3', title: 'Tab 3', content: 'Some text 3' },
-// ];
-
-export const App = () => (
-  <>
-    {/* Also requires <html class="has-navbar-fixed-top"> */}
-    <nav
-      className="navbar is-light is-fixed-top is-mobile has-shadow"
-      data-cy="Nav"
-    >
-      <div className="container">
-        <div className="navbar-brand">
-          <a href="/" className="navbar-item is-active">Home</a>
-          <a href="/tabs" className="navbar-item">Tabs</a>
-        </div>
-      </div>
-    </nav>
-
-    <div className="section">
-      <div className="container">
-        <h1 className="title">Home page</h1>
-        <h1 className="title">Tabs page</h1>
-        <h1 className="title">Page not found</h1>
-
-        <div className="tabs is-boxed">
-          <ul>
-            <li data-cy="Tab" className="is-active">
-              <a href="#/">Tab 1</a>
-            </li>
-            <li data-cy="Tab">
-              <a href="#/">Tab 2</a>
-            </li>
-            <li data-cy="Tab">
-              <a href="#/">Tab 3</a>
-            </li>
-          </ul>
+export const App = () => {
+  return (
+    <>
+      {/* Also requires <html class="has-navbar-fixed-top"> */}
+      <nav
+        className="navbar is-light is-fixed-top is-mobile has-shadow"
+        data-cy="Nav"
+      >
+        <div className="container">
+          <div className="navbar-brand">
+            <SearchLink to="/" text="Home" />
+            <SearchLink to="/tabs" text="Tabs" />
+          </div>
         </div>
+      </nav>
+
+      <div className="section">
+        <div className="container">
+          <Routes>
+            <Route
+              path="/"
+              element={<HomePage />}
+            />
+
+            <Route
+              path="home"
+              element={<Navigate to="/" replace />}
+            />
+
+            <Route
+              path="tabs"
+              element={(
+                <>
+                  <>
+                    <h1 className="title">Tabs page</h1>
+                    <Tabs tabs={tabs} />
+                  </>
+
+                  <Outlet />
+                </>
+              )}
+            >
+              <Route
+                index
+                element={(
+                  <div className="block" data-cy="TabContent">
+                    Please select a tab
+                  </div>
+                )}
+              />
+
+              <Route
+                path=":tabId"
+                element={<TabContent tabs={tabs} />}
+              />
+            </Route>
+
+            <Route
+              path="*"
+              element={<NotFound />}
+            />
+          </Routes>
```

Comment:
1. Let's create a component `Layout`. 
2. Wrap all your routes to the `layout route`. 
3. This layout component should render a `Navigation` and wrap the content into a section. You can use Outlet from react-router-dom.
4. Don't forget to remove div container and div section from app to layout

```
  <main className="section">
    <Outlet />
  </main>
```

```suggestion
<Routes>
  <Route path="/" element={<Layout />}>
    .. homepage
    ...tabs
    .. etc
</Routes>     
```